### DISCUSSION TITLE
How can we detect the image properties

### DISCUSSION THREAD

**Discussion by tonyfkchu** [dlemstra/Magick.NET#658] [Help] [answered]
Dear Expert,

Anyone can help me to convert the following CMDLine of Image Adjustment process to .Net API 

**A) Detect Image Background and Remove it**
[Convert CMDLine]
1) convert input.tiff -format "%[pixel:u.p{0,0}]" info:
2) convert input.tiff -fuzz 50% +transparent <_**bgColor result from step1**_> -scale1x1! -alpha off -format "%[pixel:u.p{0,0}]" info:
3) convert input.tiff -channel rgb -fill white -fuzz 50% -opaque <_**bgColor result from step2**_> output.tiff

**B) Detect Low Contrast Level and Enhance the Contrast** 
1) Identify -verbose input.tiff
2) Extract the Line "STANDARD DEVIATION:" under "Overall: " section
3) if the result in Step 2 lower then 40 Apply Step 4 Otherwise Step 5
4) convert input.tiff -channel RGB -level 50%,80% output.tiff
5) convert input.tiff -channel RGB -auto-level output.tiff

Thanks for any help!

Best Regards
Tony Chu

**Comment by tonyfkchu** [comment, marked as answer]
Dear Expert, 

After more study, I find a way to convert the code of Remove Background Color For **Item A** as follow:

    Public Function RemoveBG(Fuzz As Integer) As Stream
        Dim result As Stream = New MemoryStream()
        Dim bgColor = IMImage.BackgroundColor

        IMImage.ColorFuzz = New Percentage(Fuzz)
        IMImage.Settings.FillColor = bgColor
        IMImage.Opaque(bgColor, MagickColors.White)

        IMImage.Write(result)
        Return result
    End Function

**Reply by dlemstra** [reply]
A) You can get the name of the `MagickColor` that you need to call the `Transparent` method on the `image` with the following call `image.FormatExpression("%[pixel:u.p{0,0}]")`.

B) You can get the STANDARD DEVIATION with `image.Statistics().Composite().StandardDeviation`.

I hope this will help you get started.

**Reply by tonyfkchu** [reply]
I have tested the Standard Deviation with this image https://app.box.com/s/3ihciqfhbgbanzovu8kg5qiknxzhkhqy

For the API, it return the result as 6402.11455673883 

but with the identify -verbose function it just return 24.5184 as show below

  Image statistics:
    Overall:
      min: 0  (0)
      max: 255 (1)
      mean: 239.587 (0.939559)
      standard deviation: **24.5184 (0.0961508)**
      kurtosis: 46.5539
      skewness: -6.03602
      entropy: 0.428667
So what cost the difference ?

Best Regards
Tony Chu

**Reply by dlemstra** [reply]
You are using Q8 on the command line and Q16 in .NET. The Q8 version use a range of 0 to 255 and the Q16 version uses a range from 0 to 65535. Percentage wise this is around the same value.

**Comment by tonyfkchu** [comment]
I have found 1 more way to detect contrast level  with CMDLine 

magick zelda1.jpg -colorspace LAB -channel 0 -separate +channel \
\( -clone 0 -statistic maximum 5x5 \) \
\( -clone 0 -statistic minimum 5x5 \) \
\( -clone 1 -clone 2 +swap -define compose:clamp=off -compose minus -composite \) \
\( -clone 1 -clone 2 +swap -define compose:clamp=off -compose plus -composite \) \
-delete 0-2 \
+swap -define compose:clamp=off -compose divide -composite \
-scale 1x1! -format "%[fx:100*u]\%" info:

How can we convert it to API ?

Best Regards
Tony Chu

**Reply by dlemstra** [reply]
What have you tried so far?

**Comment by tonyfkchu** [comment]
So Far, I just try the following only

image.Statistics().Composite().StandardDeviation.

**Reply by dlemstra** [reply]
You have added multiple samples. I am no longer following what you want to accomplish.
